backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeBoundsProvider.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Thales.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.diagrams.components;
+
+import java.util.List;
+
+import org.eclipse.sirius.web.annotations.graphql.GraphQLNonNull;
+import org.eclipse.sirius.web.components.Element;
+import org.eclipse.sirius.web.diagrams.Position;
+import org.eclipse.sirius.web.diagrams.Size;
+
+/**
+ * Provides the bounds (Position and Size) to apply to a new node.
+ *
+ * @author fbarbin
+ */
+public class NodeBoundsProvider {
+
+    private static final int NEXT_POSITION_DELTA = 30;
+
+    // @formatter:off
+    private Position startingPosition = Position.newPosition()
+      .x(0)
+      .y(0)
+      .build();
+    // @formatter:on
+
+    private Position lastPosition;
+
+    public @GraphQLNonNull Position getNextPosition() {
```

Comment:
Why `@GraphQLNonNull`? That does not make sense here

Reply:
That is a mistake